api/misc.go

```diff
@@ -29,17 +29,41 @@ func (s *Server) eventList(apiContext *api.ApiContext) (*client.GenericCollectio
 	return toEventCollection(eventList), nil
 }
 
-func (s *Server) GenerateSupportBundle(w http.ResponseWriter, req *http.Request) error {
-	bundleFile, fileName, size, err := s.m.GenerateSupportBundle()
+func (s *Server) InitiateSupportBundle(w http.ResponseWriter, req *http.Request) error {
+	apiContext := api.GetApiContext(req)
+	nodeID, fileName, err := s.m.InitSupportBundle()
 	if err != nil {
-		return err
+		return errors.Wrap(err, "Unable to initiate Support Bundle Download")
 	}
-	defer bundleFile.Close()
+	apiContext.Write(toSBResource(nodeID, fileName))
+	return nil
+}
 
+func (s *Server) CheckSupportBundle(w http.ResponseWriter, req *http.Request) error {
+	apiContext := api.GetApiContext(req)
+	if err := s.m.CheckSBError(); err != nil {
+		s.m.SetBundleStateToReady()
```

Comment:
The support bundle state should already be updated. It shouldn't be the API call to update the support bundle state.